go/libkb/chain_link.go

```diff
@@ -735,15 +816,24 @@ func (c *ChainLink) verifyHashV1() error {
 	return nil
 }
 
-// getFixedPayload usually just returns c.unpacked.payloadJSONstr, but sometimes
+// getFixedPayload usually just returns c.unpacked.SigPayload(), but sometimes
 // it adds extra whitespace to work around server-side bugs.
 func (c ChainLink) getFixedPayload() []byte {
-	ret := c.unpacked.PayloadJSONStr()
-	if s, ok := badWhitespaceChainLinks[c.unpacked.sigID]; ok {
-		c.G().Log.Debug("Fixing payload by adding newline on link '%s': %s", c.unpacked.sigID, s)
-		ret += "\n"
+	payload, err := c.unpacked.SigPayload()
+	if err != nil {
+		return nil
 	}
-	return []byte(ret)
+	return c.fixPayload(payload, c.unpacked.sigID)
+}
+
+func (c *ChainLink) fixPayload(payload []byte, sigID keybase1.SigID) []byte {
```

Comment:
it's worth manually checking that this hack still works (by ID'ing the affected people...)

Reply:
Ok, I will.  There are also tests in the sig_chain_test.go that exercise this...

Reply:
Worked for 3 users.

Reply:
:+1: